lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1180,9 +1180,13 @@ def _resource_sku_to_capability(
                 )
             elif name == "GPUs":
                 node_space.gpu_count = int(sku_capability.value)
+                # update features list if gpu feature is supported
+                node_space.features = search_space.SetSpace[str](is_allow_set=True)
+                node_space.features.update(features.Gpu.name())
 
         # all nodes support following features
-        node_space.features = search_space.SetSpace[str](is_allow_set=True)
+        if not node_space.features:
```

Comment:
Because the features will be assigned in all code path, it's better to put the initialize line before the for loop. It's easier to understand.

Reply:
Done